src/pathfinder/aystar.cpp

```diff
@@ -127,20 +125,21 @@ AyStarStatus AyStar::Loop()
 
 /**
  * This is the function you call to run AyStar.
+ * @param max_loops Limit number of iterations (0 = infinite). Return StillBusy if limit reached.
  * @return Possible values:
  *  - #AyStarStatus::FoundEndNode
  *  - #AyStarStatus::NoPath
  *  - #AyStarStatus::StillBusy
  * @note When the algorithm is done (when the return value is not #AyStarStatus::StillBusy) #Clear() is called automatically.
  *       When you stop the algorithm halfway, you should call #Clear() yourself!
  */
-AyStarStatus AyStar::Main()
+AyStarStatus AyStar::Main(uint max_loops)
 {
 	AyStarStatus r = AyStarStatus::FoundEndNode;
-	int i = 0;
+	uint i = 0;
 	/* Loop through the OpenList
 	 *  Quit if result is no AyStarStatus::StillBusy or is more than loops_per_tick */
-	while ((r = this->Loop()) == AyStarStatus::StillBusy && (this->loops_per_tick == 0 || ++i < this->loops_per_tick)) { }
+	while ((r = this->Loop()) == AyStarStatus::StillBusy && (max_loops == 0 || ++i < max_loops)) { }
```

Comment:
I can't find a place where `loops_per_tick` is set. Nor do I find any cases where `max_loops` will be set to anything other than 0. So, is `max_loops` really needed?

Reply:
I removed all three unused limit settings `loops_per_tick`, `max_path_cost` and `max_search_nodes`.